scripts/changesummary.py

```diff
@@ -0,0 +1,540 @@
+        )
+
+        # Create a list of all parents that have been deleted in hierarchical
+        # order. When `Percent` is 1, it means that the parent is new as all
+        # children keys have been added.
+        all_deleted = (
+            parent_deleted_agg[
+                (parent_deleted_agg["Percent"] == 1)
+                & (parent_deleted_agg["Deleted"] == True)
+            ][["Parent", "NumLevels"]]
+            .sort_values("NumLevels", ascending=True)
+            .Parent.to_list()
+        )
+
+        # Go through the list of parents that have been added. If we find any
+        # keys with parents which are a substring of the parent in this list,
+        # then it means that the entire parent is new. We don't need verbose
+        # information about the children, so we replace the parent.
+        for i in range(0, len(all_added)):
+            word = all_added[i]
+            combined_docs.Parent = np.where(
+                combined_docs["Parent"].str.startswith(word), word, combined_docs.Parent
+            )
+
+        # Go through the list of parents that have been deleted. If we find any
+        # keys with parents which are a substring of the parent in this list,
+        # then it means that the entire parent is deleted. We don't need verbose
+        # information about the children, so we replace the parent.
+        for i in range(0, len(all_deleted)):
+            word = all_deleted[i]
+            combined_docs.Parent = np.where(
+                combined_docs["Parent"].str.startswith(word), word, combined_docs.Parent
+            )
+
+        # Create a new dataframe with only the keys which have changed
+        docs_diff = combined_docs[
+            combined_docs["CurrentValue"] != combined_docs["NewValue"]
+        ].copy(deep=False)
+
+        # Get the API and Version from the file name but exclude the extension.
+        api_version_string = filename.split(".")[:-1]
+        # Create columns `Name` and `Version` using the version string
+        docs_diff["Name"] = api_version_string[0]
+        docs_diff["Version"] = ".".join(api_version_string[1:])
+
+        # These conditions are used as arguments in the `np.where` function
+        # below.
+        deleted_condition = docs_diff["NewValue"].isnull()
+        added_condition = docs_diff["CurrentValue"].isnull()
+
+        # Create a new `ChangeType` column. The `np.where()` function is like a
+        # tenary operator. When the `deleted_condition` is `True`, the
+        # `ChangeType` will be `ChangeType.Deleted`. If the added_condition is
+        # `True` the `ChangeType` will be `ChangeType.Added`, otherwise the
+        # `ChangeType` will be `ChangeType.Changed`.
+        docs_diff["ChangeType"] = np.where(
+            deleted_condition,
+            ChangeType.DELETED,
+            np.where(added_condition, ChangeType.ADDED, ChangeType.CHANGED),
+        )
+
+        # Filter out keys which appear to be less important such as:
```

Comment:
"appear to be less important"

Can we be more specific than this? Perhaps "don't usually affect feature behavior" or something like that?

Reply:
Done. https://github.com/googleapis/google-api-python-client/pull/1286/commits/a27342f57eefe5c423fa12fbce43c54637ddf3d0